hudi-flink-datasource/hudi-flink/src/main/java/org/apache/hudi/table/format/FormatUtils.java

```diff
@@ -130,6 +132,7 @@ public static HoodieMergedLogRecordScanner logScanner(
         .withBasePath(split.getTablePath())
         .withLogFilePaths(split.getLogPaths().get())
         .withReaderSchema(logSchema)
+        .withInternalSchema(internalSchema)
         .withLatestInstantTime(split.getLatestCommit())
```

Comment:
Just pass around a correct readSchema is enough ~

Reply:
The same https://github.com/apache/hudi/pull/5830#discussion_r927312625

Reply:
Fixed